### PR TITLE
Add check node status - Closes #507

### LINKED ISSUE #507: Add a command to get node status
### Description

Add a command to get node status of the node that you are making request to.

```
get node status
get node status --detail
```

Child issue of #441

### PR DESCRIPTION
**Pull request by shuse2**
# Description
Add `check node status` command

### Review checklist

* The PR solves #507
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: src/commands/check_node_status.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { createCommand } from '../utils/helpers';
+import getAPIClient from '../utils/api';
+
+const description = `Gets information about a node.
+
+	Examples:
```
Comment (willclarktech) [inline src/commands/check_node_status.js]: The pattern so far for single examples is
```
Example: get node status
```

File: src/commands/check_node_status.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	- get node status
```
Comment (willclarktech) [inline src/commands/check_node_status.js]: Now `check node status`

File: src/commands/check_node_status.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+`;
+
+const forgingDescription =
+	'Additionally provides information about forging status';
```
Comment (willclarktech) [inline src/commands/check_node_status.js]: Missing full stop.

File: src/utils/tablify.js [hunk 4]
```diff
@@ -42,10 +42,15 @@ const getKeyValueObject = object => {
 		.join('\n');
 };
 
+const getKeyValueArray = array =>
+	array.some(item => typeof item !== 'object')
```
Comment (willclarktech) [inline src/utils/tablify.js]: Surely we want to use `getKeyValueObject` if any of the members are objects even if some aren't?

File: test/specs/commands/check_node_status.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+							then.itShouldNotCallNodeGetForgingStatus,
+						);
+						Then(
+							'it should resolves to an object which does not have key equal to "forgingStatus"',
+							then.itShouldResolvesToAnObjectWhichDoesNotHaveKeyEqualTo,
+						);
+					},
+				);
+				Given(
+					'a boolean option "forging" set to "true"',
```
Comment (willclarktech) [inline test/specs/commands/check_node_status.js]: Actually you don't need quotes around `true`.

File: test/specs/commands/check_node_status.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+						Then(
+							'it should call node getStatus',
+							then.itShouldCallNodeGetStatus,
+						);
+						Then(
+							'it should not call node getForgingStatus',
+							then.itShouldNotCallNodeGetForgingStatus,
+						);
+						Then(
+							'it should resolves to an object which does not have key equal to "forgingStatus"',
```
Comment (willclarktech) [inline test/specs/commands/check_node_status.js]: OK, but what *should* it have?

File: test/specs/commands/check_node_status.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+										Then(
+											'it should call node getForgingStatus',
+											then.itShouldCallNodeGetForgingStatus,
+										);
+										Then(
+											'it should call node getForgingStatus',
+											then.itShoulHaveErrorMessageForForgingStatus,
+										);
+										Then(
+											'it should resolves to an object which has key equal to "forgingStatus"',
```
Comment (willclarktech) [inline test/specs/commands/check_node_status.js]: Anything else?

File: test/specs/commands/check_node_status.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+								Then(
+									'it should call node getStatus',
+									then.itShouldCallNodeGetStatus,
+								);
+								Then(
+									'it should call node getForgingStatus',
+									then.itShouldCallNodeGetForgingStatus,
+								);
+								Then(
+									'it should resolves to an object which has key equal to "forgingStatus"',
```
Comment (willclarktech) [inline test/specs/commands/check_node_status.js]: Anything else?

File: test/steps/api/3_then.js [hunk 9]
```diff
@@ -73,3 +73,19 @@ export function theLiskAPIInstanceShouldHaveCurrentNodeEqualTo() {
 	const node = getFirstQuotedString(this.test.title);
 	return expect(liskAPIInstance.currentNode).to.equal(node);
 }
+
+export function itShouldResolvesToAnObjectWhichHaveKeyEqualTo() {
```
Comment (willclarktech) [inline test/steps/api/3_then.js]: `Resolve`

File: test/steps/api/3_then.js [hunk 10]
```diff
@@ -73,3 +73,19 @@ export function theLiskAPIInstanceShouldHaveCurrentNodeEqualTo() {
...
+	const { returnValue } = this.test.ctx;
+	const key = getFirstQuotedString(this.test.title);
+	return expect(returnValue)
+		.to.eventually.be.an('object')
+		.and.have.property(key);
+}
+
+export function itShouldResolvesToAnObjectWhichDoesNotHaveKeyEqualTo() {
```
Comment (willclarktech) [inline test/steps/api/3_then.js]: `Resolve`

File: test/specs/commands/check_node_status.js (hunk 6 above)
Comment (shuse2) [reply test/specs/commands/check_node_status.js]: Other keys are strictly related to the API response, so I don't think it's good idea to check them

File: src/utils/tablify.js (hunk 4 above)
Comment (shuse2) [reply src/utils/tablify.js]: At this point, we expect all the elements of array have the same shape

File: src/commands/check_node_status.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { createCommand } from '../utils/helpers';
+import getAPIClient from '../utils/api';
+
+const description = `Gets information about a node.
+
+	Examples: check node status
```
Comment (willclarktech) [inline src/commands/check_node_status.js]: `Example:`

File: src/utils/tablify.js (hunk 4 above)
Comment (willclarktech) [reply src/utils/tablify.js]: Why not make it future-proof though?

File: test/steps/queries/3_then.js [hunk 12]
```diff
@@ -34,3 +33,31 @@ export function itShouldResolveToAnArrayOfQueryResults() {
...
+	const { liskAPIInstance } = this.test.ctx;
+	return expect(liskAPIInstance.node.getForgingStatus).to.be.calledOnce;
+}
+
+export function itShouldNotCallNodeGetForgingStatus() {
+	const { liskAPIInstance } = this.test.ctx;
+	return expect(liskAPIInstance.node.getForgingStatus).not.to.be.called;
+}
+
+export function itShoulHaveErrorMessageForForgingStatus() {
```
Comment (willclarktech) [inline test/steps/queries/3_then.js]: `itShould`

File: test/steps/queries/3_then.js [hunk 13]
```diff
@@ -34,3 +33,31 @@ export function itShouldResolveToAnArrayOfQueryResults() {
...
+	const { returnValue, errorMessage } = this.test.ctx;
+	return expect(returnValue)
+		.to.eventually.be.an('object')
+		.and.have.property('forgingStatus')
+		.eql(errorMessage);
```
Comment (willclarktech) [inline test/steps/queries/3_then.js]: `.equal`

**Pull request merged**